shared/chat/manage-channels/container.js

```diff
@@ -88,27 +96,21 @@ const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, routePath, routePro
       oldChannelState: ChannelMembershipState,
       nextChannelState: ChannelMembershipState
     ) => {
-      const channelsToChange = pickBy(
-        nextChannelState,
-        (inChannel: boolean, channelname: string) => inChannel !== oldChannelState[channelname]
-      )
-      dispatch(TeamsGen.createSaveChannelMembership({teamname, channelState: channelsToChange}))
+      dispatch(createSaveChannelMembership(teamname, oldChannelState, nextChannelState))
+      dispatch(navigateUp())
     },
-    _onPreview: (conversationIDKey: ChatTypes.ConversationIDKey, previousPath?: string[]) => {
-      dispatch(Chat2Gen.createSelectConversation({conversationIDKey, reason: 'preview'}))
+    _onView: (
+      oldChannelState: ChannelMembershipState,
+      nextChannelState: ChannelMembershipState,
+      channel: string,
+      conversationIDKey: ChatTypes.ConversationIDKey
+    ) => {
+      const selected = nextChannelState[channel]
+      dispatch(createSaveChannelMembership(teamname, oldChannelState, nextChannelState))
       dispatch(
-        navigateTo([
-          chatTab,
-          {
-            selected: ChatTypes.conversationIDKeyToString(conversationIDKey),
-            props: {previousPath: previousPath || null},
-          },
-        ])
+        Chat2Gen.createSelectConversation({conversationIDKey, reason: selected ? 'manageView' : 'preview'})
```

Comment:
the reason is selected from `nextChannelState[channel]`, so it races with actually applying that state; a bit of an edge case, but at worst only affects the display of the 'preview' blurb.